### ISSUE TITLE
Issue when creating and saving a new playlist from a selection in the Playlist Editor

### ISSUE THREAD

**Issue by touwys** [Borewit/listFix#193]
**Steps:**

_Playlist Editor > Right-click > Playlist Context Menu > Select "Create New Playlist With Selected Items":_

Produces the desired playlist in a new tab named "Untitled-0.m3u8", "Untitled-1.m3u8", etc.

_**but:**_

1. The file format of the playlist in the editor — the one from which the new playlist file is produced, — is M3U. One would expect the same extension to apply to the newly created playlist by default?

2. To save the new playlist by clicking on the Save button, i.e. the diskette icon on the toolbar, produces an unidentifiable flash of "something" on the screen. It does not produce the expected Save Dialogue box, which usually contains the various saving options such as, for example, the preferred name, the location, nor the desired file extension.

—

**Comment by Borewit** [comment]
Current behavior is indeed that by the default a `.m3u8` playlist is created.

**Comment by Borewit** [comment]
> To save the new playlist by clicking on the Save button, i.e. the diskette icon on the toolbar, produces an unidentifiable flash of "something" on the screen. It does not produce the expected Save Dialogue box, which usually contains the various saving options such as, for example, the preferred name, the location, nor the desired file extension.

The current behavior is that a new playlist receives a name auto generated. When the save button is generated is simply saves to disks. Should be raised to a different issue if you suggest to this different, as is has nothing to do with the different file-type which is generated mentioned in 1.

**Comment by Borewit** [comment]
Regarding the second part:

The save button just saves the new playlist, the flash is the save progress, which seems like flash,  as it does not take very long.

> It does not produce the expected Save Dialogue box, which usually contains the various saving options such as, for example, the preferred name, the location, nor the desired file extension.

Because it a temporary file, it should show the  _Save as_ dialog. Did I break that functionality? I can imagine I did. What if we save directly when creating new playlist, to keep things simpler? To avoid we to keep track which playlist have been saved in a temporary way, clean op temporary files, etc?

**Comment by touwys** [comment]
> The save button just saves the new playlist, the flash is the save progress, which seems like flash, as it does not take very long.

I guessed that much, but where does this temporary file get saved to? Location? HDD or memory?

> Because it a temporary file, it should show the Save as dialog. Did I break that functionality? I can imagine I did. What if we save directly when creating new playlist, to keep things simpler? To avoid we to keep track which playlist have been saved in a temporary way, clean op temporary files, etc?

I do not have a clear picture of exactly you have in mind here. My only concern is that saving a newly created, untitled playlist at the moment, operates without any visible clues — i.e. those user-input dialogues that will usually seek user approval before the file gets saved. Therefore, if you are for supporting a "Save as" dialogue, I'm certainly all for it too.

In summary, I would like to see support for a "Save as" dialogue that appears when the save diskette icon is clicked, allowing the user to exercise those options which are available to them — of which the save location, file name and the file extension appears to be the most prominent.
